compiler_tester/src/workflow.rs

```diff
@@ -0,0 +1,29 @@
+//!
+//! Compiler tester workflows
+//!
+
+///
+/// Describes sets of actions that compiler tester is able to perform.
+///
+#[derive(Debug)]
+pub enum Workflow {
+    /// Only build tests but not execute them.
+    BuildOnly,
+    /// Build and execute tests.
+    BuildAndRun,
+}
+use std::str::FromStr;
+
+// any error type implementing Display is acceptable.
+type ParseError = &'static str;
+
+impl FromStr for Workflow {
+    type Err = ParseError;
```

Comment:
Simply put `&'static str` here. It is common practice to return the arg as error.